tests/test_web/test_messaging.py

```
@@ -0,0 +1,164 @@
+        messaging.Envelope('topic://VirtualTopic.eng.star_wars2', msg_one),
+        messaging.Envelope('topic://VirtualTopic.eng.star_wars', msg_two),
+    ]
+    messaging.send_messages(envelopes)
+
+    mock_bc.assert_called_once_with(
+        urls=['amqps://message-broker:5671'], timeout=30, ssl_domain=mock_gsd.return_value
+    )
+    # Verify that even though three messages were sent, only two senders were created since only
+    # two unique addresses were used
+    assert mock_connection.create_sender.call_count == 2
+    # Ensure the order is respected
+    mock_connection.create_sender.assert_has_calls(
+        (
+            mock.call('topic://VirtualTopic.eng.star_wars'),
+            mock.call('topic://VirtualTopic.eng.star_wars2'),
+        )
+    )
+    assert mock_sender_one.send.call_count == 2
+    mock_sender_one.send.assert_has_calls(
+        (mock.call(msg_one, timeout=30), mock.call(msg_two, timeout=30))
+    )
+    mock_sender_two.send.assert_called_once_with(msg_one, timeout=30)
+    mock_connection.close.assert_called_once_with()
+
+
+@pytest.mark.parametrize(
+    'missing_config',
+    (None, 'IIB_MESSAGING_BUILD_STATE_DESTINATION', 'IIB_MESSAGING_BATCH_STATE_DESTINATION'),
+)
+@pytest.mark.parametrize('request_state', ('complete', 'in_progress'))
+@pytest.mark.parametrize('new_batch_msg', (True, False))
+@mock.patch('iib.web.messaging.send_messages')
+def test_send_message_for_state_change(
+    mock_sm, new_batch_msg, request_state, missing_config, app, db, minimal_request_add
+):
+    minimal_request_add.add_state(request_state, 'For some reason')
+    db.session.add(minimal_request_add)
+    db.session.commit()
+    if missing_config:
+        app.config.pop(missing_config)
+
+    messaging.send_message_for_state_change(minimal_request_add, new_batch_msg)
+
+    request_expected = True
+    batch_expected = True
+
+    if not new_batch_msg and request_state == 'in_progress':
```

Consider combining the `parametrize` calls so you can provide the expectation as a parameter to the test function. Relying on conditionals for setting expectations can hide errors.